snapshot/pkg/volume/openebs/processor.go

```diff
@@ -0,0 +1,312 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openebs
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	mApiv1 "github.com/kubernetes-incubator/external-storage/openebs/pkg/v1"
+	mayav1 "github.com/kubernetes-incubator/external-storage/openebs/types/v1"
+
+	"github.com/golang/glog"
+	crdv1 "github.com/kubernetes-incubator/external-storage/snapshot/pkg/apis/crd/v1"
+	"github.com/kubernetes-incubator/external-storage/snapshot/pkg/cloudprovider"
+	"github.com/kubernetes-incubator/external-storage/snapshot/pkg/volume"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	openEBSPersistentDiskPluginName = "openebs"
+)
+
+type openEBSPlugin struct {
+	mApiv1.OpenEBSVolume
+}
+
+var _ volume.Plugin = &openEBSPlugin{}
+
+// RegisterPlugin registers the volume plugin
+func RegisterPlugin() volume.Plugin {
+	return &openEBSPlugin{}
+}
+
+func init() {
+	// GetMayaService get the maya-service endpoint
+	_ = GetMayaService()
+}
+
+// GetPluginName gets the name of the volume plugin
+func GetPluginName() string {
+	return "openebs"
+}
+
+func (h *openEBSPlugin) Init(_ cloudprovider.Interface) {
+}
+
+func (h *openEBSPlugin) SnapshotCreate(pv *v1.PersistentVolume, tags *map[string]string) (*crdv1.VolumeSnapshotDataSource, *[]crdv1.VolumeSnapshotCondition, error) {
+	spec := &pv.Spec
+	if spec == nil || spec.ISCSI == nil {
+		return nil, nil, fmt.Errorf("invalid PV spec %v", spec)
+	}
+
+	// GetMayaService get the maya-service endpoint
+	//err := GetMayaService()
+	//if err != nil {
+	//	return nil, nil, err
+	//}
+
+	//snapshotName := &tags["kubernetes.io/created-for/snapshot/name"]
+	snapshotName := createSnapshotName(pv.Name)
```

Comment:
shouldn't we pass the snapname here as well to generate a unique snapname?

Reply:
Added the snapshot name as a part of unique snapshot name